added documentation for connecting firefly to a remote fabric chaincode (#908)
hyperledger/firefly: docs/tutorials/fabric_remote.md

- Download Docker images for all of the components of the Supernode
- Set up configuration between all the components
- Deploy FireFly's `BatchPin` smart contract to our remote fabric network
- Register an identity for each member and node in our remote fabric network
- It will probably also initialize a new blockchain and blockchain node running inside a container and also deploy an `ERC-1155` token smart contract to that local fabric network. This behaviour can be ignored.

After your stack finishes starting it will print out the links to each member's UI and the Sandbox for that node. For this guide, it's just one member.

```
Web UI for member '0': http://127.0.0.1:5000/ui
Sandbox UI for member '0': http://127.0.0.1:5108

To see logs for your stack run:

ff logs my-remote-fabric

```

## Integrate your FireFly stack with your Fabric chaincode 

Once your stack is up and running, you can define and broadcast FireFly Interface Document for your remote fabric chaincode. You can refer to [Broadcast the Contract Interface(Fabric)](http://localhost:4000/firefly/tutorials/custom_contracts/fabric.html#broadcast-the-contract-interface) guide for this.

> You can also now create an HTTP API for your fabric chaincode which will help you easily query/invoke all your chaincode methods with your organizational context that you set up in fabconnect. You can refer to [Create an HTTP API for the contract(Fabric)](http://localhost:4000/firefly/tutorials/custom_contracts/fabric.html#broadcast-the-contract-interface) guide for this.

Reviewer: ```suggestion
You can also now create an HTTP API for your fabric chaincode which will help you easily query/invoke all your chaincode methods with your organizational context that you set up in fabconnect. You can refer to [Create an HTTP API for the contract(Fabric)](./custom_contracts/fabric.html#broadcast-the-contract-interface) guide for this.
```

These need to be relative links so they will link correctly when the docs site is published to GitHub pages.

> To view the OpenAPI spec for your contract, or to submit transactions, query for states and listen for events, you can further refer to [Work with Hyperledger Fabric chaincodes](http://localhost:4000/firefly/tutorials/custom_contracts/fabric.html#work-with-hyperledger-fabric-chaincodes) guide.

Reviewer: ```suggestion
To view the OpenAPI spec for your contract, or to submit transactions, query for states and listen for events, you can further refer to [Work with Hyperledger Fabric chaincodes](./custom_contracts/fabric.html#work-with-hyperledger-fabric-chaincodes) guide.
```

---
layout: default
title: Work with remote fabric network
parent: pages.tutorials
nav_order:
---

# Work with remote Hyperledger Fabric Network

> 

Reviewer: ```suggestion
```
Removing this extra line of whitespace will achieve the desired effect of not having the page title in the page's table of contents below

{: .no_toc }

> This guide describes the steps to use FireFly to interact with a chaincode deployed to an external Hyperledger Fabric blockchain network (not created by Firefly) in order to submit transactions and query for states.

Reviewer: I think adding a note at the top of this tutorial to set a bit more context would be helpful to people. While this is a fantastic guide for helping people connect a local development environment to a pre-existing Fabric network, we don't want people to get the idea that the FireFly CLI should be used for setting up production networks. I might suggest something like this:

```suggestion
The FireFly CLI makes it quick and easy to create an entire FireFly development environment on your local machine, including a new blockchain from scratch. However, sometimes a developer may want to connect their local FireFly development environment to a Fabric network that already exists on their machine or elsewhere. This guide describes the steps to connect FireFly to an external Hyperledger Fabric blockchain network (not created by the FireFly CLI), including interacting with a custom chaincode in order to submit transactions and query for state.
```

### Start your stack

To start your stack now, simply run:

```
ff start my-remote-fabric
```

This may take a minute or two and in the background the FireFly CLI will do the following for you:

- Download Docker images for all of the components of the Supernode
- Set up configuration between all the components
- Deploy FireFly's `BatchPin` smart contract to our remote fabric network
- Register an identity for each member and node in our remote fabric network
- It will probably also initialize a new blockchain and blockchain node running inside a container and also deploy an `ERC-1155` token smart contract to that local fabric network. This behaviour can be ignored.

After your stack finishes starting it will print out the links to each member's UI and the Sandbox for that node. For this guide, it's just one member.

```
Web UI for member '0': http://127.0.0.1:5000/ui
Sandbox UI for member '0': http://127.0.0.1:5108

To see logs for your stack run:

ff logs my-remote-fabric

```

> ## Define and Broadcast FireFly Interface Document for your chaincode

Reviewer: It looks like most of this section and the rest of this guide are copied from this page: https://hyperledger.github.io/firefly/tutorials/custom_contracts/fabric.html

I think we can have a short description of what you need to do next here, but then link to that page for the details. That way we only have one page we need to update / add to when we need to talk about using custom contracts with Fabric networks.

If you feel there is information missing from that tutorial (I see you added some detail about the Sandbox here which is great!) please feel free to update the other tutorial as well.

Author: Addressed.
